elasticdl/python/worker/worker.py

```diff
@@ -679,12 +669,10 @@ def _run_model_call_before_training(self, features):
         ):
             self._non_embed_vars[var.name] = var
 
-        if not self._var_created:
-            if self._use_multi_ps:
-                self.init_ps_var_partition()
-            else:
-                self.report_variable()
-            self._var_created = True
+        self._var_created = True
+
+        if self._use_multi_ps:
+            self.init_ps_var_partition()
```

Comment:
Is it heavy to call `self.init_ps_var_partition` multiple times?

Reply:
It will only be called for once. 

`init_ps_var_partition` is called inside `_run_model_call_before_training`. The `_run_model_call_before_training` method is only called once.